leafmap/bokehmap.py

```diff
@@ -67,7 +68,12 @@ def _repr_mimebundle_(self, **kwargs):
             os.environ["OUTPUT_NOTEBOOK"] = "True"
         show(self.figure)
 
-    def add_basemap(self, basemap="OpenStreetMap", retina=True, **kwargs):
+    def add_basemap(
+        self,
+        basemap: Optional[str] = "OpenStreetMap",
```

Comment:
Are we sure that we don't want the typing to raise an error on cases like:

```python3
map.add_basemap(None, None)
```

In these cases, the defaults would **not** be used. `None`s would be 
used in the function body. There are a lot of cases like this in the PR where None is allowed, but would cause a runtime exception

Reply:
Maybe these `Optional[T]`s should be replaced by `T`? Signaling to the user that `None` is not a valid parameter, avoiding some runtime exceptions

Reply:
We can improve the function body to raise errors when basemap is None. Indicating this error in the typing would also be helpful.

Reply:
Should we keep at Optional[str] and then handle `None` in code itself ?